lib/linked_list.rb

```diff
@@ -1,45 +1,83 @@
 require_relative 'node'
 
 class LinkedList
-  attr_reader :head
+  attr_reader :head, :tail
 
   def initialize
     @head = nil
+    @tail = tail
   end
 
-  # Time complexity - ?
-  # Space complexity - ?
+  # Time complexity - O(1)
+  # Space complexity - O(1)
   def add_first(data)
+    first = Node.new(data)
 
+    if @head
+      first.next = @head
+    end
+
+    @head = first
   end
 
-  # Time complexity - ?
-  # Space complexity - ?
+  # Time complexity - O(1)
+  # Space complexity - O(1)
   def get_first
+    if @head.nil?
+      return nil
+    end
+
+    return @head.data
 
   end
 
-  # Time complexity - ?
-  # Space complexity - ?
+  # Time complexity - O(1)
+  # Space complexity - O(1)
   def length
```

Comment:
How is the time complexity O(1)?  You have a loop.

Reply:
ah! yes, i changed this and just forgot to resubmit. it would be O(n)